src/cli/init.rs

```diff
@@ -334,12 +334,16 @@ pub async fn execute(args: Args) -> miette::Result<()> {
 
             // Create a 'pyproject.toml' manifest
         } else if pyproject {
+            // Python package names cannot contain '-', so we replace them with '_'
+            let pypi_package_name = default_name.replace("-", "_");
```

Comment:
Perhaps better to parse the name as PackageName and use the dist info name.

Reply:
Changed it!